### ISSUE TITLE
Snap package - lxc stop - failed to connect to monitor socket

### ISSUE THREAD

**Issue by mikescott** [lxc/lxd#3032]
# Issue description
I'm using the Snap package of LXD, in Debian 9.0 'Stretch' which generally is working well but stopping a container produces the following error:

```
error: shutting down the container failed
Try `lxc info --show-log  container1` for more info
```

The container does stop despite this error.

# lxc info
```
config: {}
apiextensions:
- storage_zfs_remove_snapshots
- container_host_shutdown_timeout
- container_syscall_filtering
- auth_pki
- container_last_used_at
- etag
- patch
- usb_devices
- https_allowed_credentials
- image_compression_algorithm
- directory_manipulation
- container_cpu_time
- storage_zfs_use_refquota
- storage_lvm_mount_options
- network
- profile_usedby
- container_push
- container_exec_recording
- certificate_update
- container_exec_signal_handling
- gpu_devices
- container_image_properties
- migration_progress
- id_map
- network_firewall_filtering
- network_routes
apistatus: stable
apiversion: "1.0"
auth: trusted
public: false
environment:
  addresses: []
  architectures:
  - x86_64
  - i686
  certificate: |
    -----BEGIN CERTIFICATE-----
[removed]
    -----END CERTIFICATE-----
  certificatefingerprint: 803bfdbbc17f0cd2b3680d489c7e146301dd757aa3f9c9b62b725fb9220bd3fd
  driver: lxc
  driverversion: 2.0.7
  kernel: Linux
  kernelarchitecture: x86_64
  kernelversion: 4.9.0-1-amd64
  server: lxd
  serverpid: 26838
  serverversion: "2.8"
  storage: dir
  storageversion: ""
```

# Steps to reproduce

 1. Fresh install of Stretch RC2 (tested on two different hosts)
 2. `apt-get update && apt-get upgrade`
 3. `apt-get install snapd`
 4. `snap install lxd`
 5. `lxd init --auto`
 6. `lxc launch images:debian/stretch container1`
 7. `lxc stop container1`

# Output of lxc info --show-log  container1
```
Name: container1
Remote: unix:/var/snap/lxd/common/lxd/unix.socket
Architecture: x86_64
Created: 2017/03/03 13:43 UTC
Status: Stopped
Type: persistent
Profiles: default

Log:

            lxc 20170307124759.832 ERROR    lxc_monitor - monitor.c:lxc_monitord_spawn:369 - Failed to exec lxc-monitord.
            lxc 20170307124800.580 ERROR    lxc_monitor - monitor.c:lxc_monitor_open:218 - Failed to connect to monitor socket. Retrying in 10 ms.
            lxc 20170307124800.682 ERROR    lxc_monitor - monitor.c:lxc_monitor_open:218 - Failed to connect to monitor socket. Retrying in 50 ms.
            lxc 20170307124800.118 ERROR    lxc_monitor - monitor.c:lxc_monitor_open:218 - Failed to connect to monitor socket. Retrying in 100 ms.
            lxc 20170307124800.218 ERROR    lxc_monitor - monitor.c:lxc_monitor_open:223 - Failed to connect to monitor socket: Connection refused.
            lxc 20170307124803.808 WARN     lxc_commands - commands.c:lxc_cmd_rsp_recv:172 - Command get_cgroup failed to receive response: Connection reset by peer.
```
# Relevant ps output
```
root     26515  0.0  0.4 455120  9440 ?        Ssl  Mar03   1:16 /usr/lib/snapd/snapd
root     26815  0.0  0.0   4508   280 ?        Ss   Mar03   0:00 /bin/sh /snap/lxd/976/wrappers/daemon.start
root     26835  0.0  0.0 160920   136 ?        Sl   Mar03   0:00  \_ lxcfs /var/snap/lxd/common/var/lib/lxcfs -p /var/snap/lxd/common/lxcfs.pid
root     26838  3.6  0.9 824036 18404 ?        Sl   Mar03 210:17  \_ lxd --logfile /var/snap/lxd/common/lxd/logs/lxd.log
```

**Comment by stgraber** [comment]
Interesting, I remember seeing this happen once. I'll take a look.

**Comment by stgraber** [comment]
This turned out to be a snap packaging issue which I've now fixed in the edge channel.
The candidate channel will get the fix tomorrow when we release LXD 2.11 and assuming that all goes well this time, the stable channel will get the fix on Thursday.

https://github.com/lxc/lxd-pkg-ubuntu/commit/87c043cada69319fa9ac9e0ba0b27b5a34a98dd0

**Comment by mikescott** [comment]
Hi @stgraber I updated to LXD 2.11 today and can confirm my containers shut down without error. Thanks for fixing so quickly 👍
